core/src/mindustry/world/blocks/defense/OverdriveProjector.java

```diff
@@ -88,17 +96,20 @@ public void updateTile(){
                 phaseHeat = Mathf.lerpDelta(phaseHeat, Mathf.num(cons.optionalValid()), 0.1f);
             }
 
-            if(timer(timerUse, useTime) && efficiency() > 0 && consValid()){
-                consume();
-            }
-
             if(charge >= reload){
                 float realRange = range + phaseHeat * phaseRangeBoost;
-                float realBoost = (speedBoost + phaseHeat * speedBoostPhase) * efficiency();
 
                 charge = 0f;
-                indexer.eachBlock(this, realRange, other -> true, other -> other.applyBoost(realBoost, reload + 1f));
+                indexer.eachBlock(this, realRange, other -> true, other -> other.applyBoost(realBoost(), reload + 1f));
             }
+
+            if(timer(timerUse, useTime) && efficiency() > 0 && consValid()){
+                consume();
+            }
+        }
+
+        public float realBoost(){
+            return consValid() ? (speedBoost + phaseHeat * speedBoostPhase) * efficiency() : 0f;
```

Comment:
Any particular reason why you added `consValid()` here?

Reply:
yes, without that the overdrive dome will show a full boost bar if it has power without the required items present

Reply:
(which is also why i moved the consume down, to prevent this function from returning 0 when its using its last set of items)